### PR TITLE
office-js-preview, contextual tabs APIs

### PR DESCRIPTION
**Pull request by Rick-Kirkham**
Please fill in this template.

- [ x] Use a meaningful title for the pull request. Include the name of the package modified.
- [ x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x ] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x ] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [ ] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If changing an existing definition:
- [ x] Provide a URL to documentation or source code which provides context for the suggested changes: https://docs.microsoft.com/en-us/javascript/api/office/office.tab?view=common-js
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`. If for reason the any rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]` and not for whole package so that the need for disabling can be reviewed.

### CODE REVIEW

**Comment by Rick-Kirkham** [comment]
Don't merge yet.

File: types/office-js-preview/index.d.ts [hunk 1]
```diff
@@ -580,12 +580,24 @@ declare namespace Office {
         ): Promise<RemoveEventListener>;
     }
     /**
-     * An interface that contains all the functionality provided to manage the state of the OFfice ribbon.
+     * An interface that contains all the functionality provided to manage the state of the Office ribbon.
      */
     interface Ribbon {
+        /**
+         * Registers a custom contextual tab with Office and defines the tab's controls.
+         * This method only requests that the tab be registered. The actual registration is controlled by the Office application and may not be complete when the returned Promise object is resolved.
```
Comment (AlexJerabek) [inline types/office-js-preview/index.d.ts:588]: Nit: `Promise` should have code ticks around it.
```suggestion
         * This method only requests that the tab be registered. The actual registration is controlled by the Office application and may not be complete when the returned `Promise` object is resolved.
```

Comment (Rick-Kirkham) [reply types/office-js-preview/index.d.ts:588]: done

Comment (Rick-Kirkham) [reply types/office-js-preview/index.d.ts:588]: Fixed

**Pull request merged**